client/src/main/java/io/crate/client/CrateClient.java

```diff
@@ -110,14 +121,28 @@ public CrateClient() {
     public CrateClient(String... servers) {
         this();
         for (String server : servers) {
-            String[] parts = server.split(":");
-            String host = parts[0];
-            Integer port = 4300;
-            if (parts.length == 2) {
-                port = Integer.parseInt(parts[1]);
-            }
-            internalClient.addTransportAddress(new InetSocketTransportAddress(host, port));
+            TransportAddress transportAddress = tryCreateTransportFor(server);
+            if(transportAddress != null)
+                internalClient.addTransportAddress(transportAddress);
+        }
+    }
+
+    protected TransportAddress tryCreateTransportFor(@Nullable String server) {
```

Comment:
the method should be annotated with `@Nullable`. `server` cannot be `null`, right?